File: Segundo/PDOO/deepSpace/src/deepspace/BetaPowerEfficientSpaceStation.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author carmenc
 */
public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation{
    private static float EXTRAEFFICIENCY=1.2f;
    private Dice dice;
    
    public BetaPowerEfficientSpaceStation(SpaceStation station){
        super(station);
        dice=new Dice();
    }
    
    @Override
    public float fire(){
        if(dice.extraEfficiency()) return super.fire()*EXTRAEFFICIENCY;
        else{
            return super.fire();
        }
    }
    
    @Override
    public SpaceStationToUI getUIversion(){
        return new BetaPowerEfficientSpaceStationToUI(this);
    }
}

File: Segundo/PDOO/ruby p4/BetaPowerEfficientSpaceStation.rb
require_relative'PowerEfficientSpaceStation'
require_relative'Dice'
module Deepspace
    class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
        EXTRAEFFICIENCY=1.2
        def fire
            dice=Dice.new
            if(dice.extraEfficiency)
                super
            else #esta mal->arreglar
                normal=SpaceStation.new(this)
                normal.fire
            end
        end
    end
end